File: sandbox/algoritms/recurency/maxInList.js
// find the largest number in the list

function maxInList(list) {
  if (list.length === 0) {
    return null;
  } else if (list.length === 1) {
    return list[0];
  } else {
    let max = list[0];
    let rest = maxInList(list.slice(1));
    return max > rest ? max : rest;
  }
}

console.log(maxInList([2, 4, 100, 3, 6])); // 6

File: Projects/Testtrack/algoritms/recurency/max_in_list.py
def max_in_list(arr: list) -> int:
    if len(arr) == 0:
        return None
    elif len(arr) == 1:
        return arr[0]
    else:
        max = arr[0]
        rest = max_in_list(arr[1:])

        return max if max > rest else rest


if __name__ == "__main__":
    print(max_in_list([2, 3, 100, 3, 12, 25, 101, 32]))
